app/src/main/java/com/sunlightlabs/congress/services/CommitteeService.java

```diff
@@ -1,117 +1,235 @@
 package com.sunlightlabs.congress.services;
 
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
+import org.apache.http.impl.cookie.DateParseException;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import com.sunlightlabs.congress.models.Bill;
 import com.sunlightlabs.congress.models.Committee;
 import com.sunlightlabs.congress.models.CongressException;
 import com.sunlightlabs.congress.models.Legislator;
 
 public class CommitteeService {
-	
-	public static String[] basicFields = new String[] {
-		"committee_id", "chamber", "name",
-		"parent_committee_id", "subcommittee"
-	};
-	
-	public static Committee find(String id) throws CongressException {
-		String[] fields = new String[] {
-			"committee_id", "chamber", "name",
-			"parent_committee_id", "subcommittee",
-			"members"
-		};
-		Map<String,String> params = new HashMap<String,String>();
-		params.put("committee_id", id);
-		return committeeFor(Congress.url("committees", fields, params));
-	}
 
-	public static List<Committee> forLegislator(String bioguideId) throws CongressException {
-		Map<String,String> params = new HashMap<String,String>();
-		params.put("member_ids", bioguideId);
-		return committeesFor(Congress.url("committees", basicFields, params, 1, Congress.MAX_PER_PAGE));
-	}
-	
-	public static List<Committee> getSubcommitteesFor(String committeeId) throws CongressException {
-		Map<String,String> params = new HashMap<String,String>();
-		params.put("subcommittee", "true");
-		params.put("parent_committee_id", committeeId);
-		return committeesFor(Congress.url("committees", basicFields, params, 1, Congress.MAX_PER_PAGE));
-	}
-	
-	public static List<Committee> getAll(String chamber) throws CongressException {
-		Map<String,String> params = new HashMap<String,String>();
-		params.put("chamber", chamber);
-		params.put("subcommittee", "false");
-		params.put("order", "name__asc");
-		return committeesFor(Congress.url("committees", basicFields, params, 1, Congress.MAX_PER_PAGE));
+    public static Committee find(String committee_id) throws CongressException {
+        return find(committee_id, null);
+    }
+
+    // /{congress}/{chamber}/committees/{committee_id}.json
+    // /{congress}/{chamber}/committees/{committee_id}/subcommittees/{subcommittee_id}.json
+	public static Committee find(String committee_id, String subcommittee_id) throws CongressException {
+        String chamber;
+        if (committee_id.toLowerCase().startsWith("h"))
+            chamber = "house";
+        else if (committee_id.toLowerCase().startsWith("s"))
+            chamber = "senate";
+        else
+            chamber = "joint";
```

Comment:
Relying on committee IDs always starting with the first letter of their chamber.